From 890d039f067d52a0ab77b561d40da13f0c135634 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 3 Aug 2017 11:24:43 -0700
Subject: [PATCH] initsendmsg: Return NULL on error

Fixes a clang error

sendmsg.c:129:9: error: non-void function 'initsendmsg' should return a value [-Wreturn-type]

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 twisted/python/sendmsg.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/twisted/python/sendmsg.c b/twisted/python/sendmsg.c
index 9f2fd8c..9c080e1 100644
--- a/twisted/python/sendmsg.c
+++ b/twisted/python/sendmsg.c
@@ -126,7 +126,7 @@ PyMODINIT_FUNC initsendmsg(void) {
     module = Py_InitModule3("sendmsg", sendmsg_methods, sendmsg_doc);
 
     if (!module) {
-        return;
+        return NULL;
     }
 
     /*
@@ -135,39 +135,39 @@ PyMODINIT_FUNC initsendmsg(void) {
     */
 
     if (-1 == PyModule_AddIntConstant(module, "SCM_RIGHTS", SCM_RIGHTS)) {
-        return;
+        return NULL;
     }
 
 
     /* BSD, Darwin, Hurd */
 #if defined(SCM_CREDS)
     if (-1 == PyModule_AddIntConstant(module, "SCM_CREDS", SCM_CREDS)) {
-        return;
+        return NULL;
     }
 #endif
 
     /* Linux */
 #if defined(SCM_CREDENTIALS)
     if (-1 == PyModule_AddIntConstant(module, "SCM_CREDENTIALS", SCM_CREDENTIALS)) {
-        return;
+        return NULL;
     }
 #endif
 
     /* Apparently everywhere, but not standardized. */
 #if defined(SCM_TIMESTAMP)
     if (-1 == PyModule_AddIntConstant(module, "SCM_TIMESTAMP", SCM_TIMESTAMP)) {
-        return;
+        return NULL;
     }
 #endif
 
     module = PyImport_ImportModule("socket");
     if (!module) {
-        return;
+        return NULL;
     }
 
     sendmsg_socket_error = PyObject_GetAttrString(module, "error");
     if (!sendmsg_socket_error) {
-        return;
+        return NULL;
     }
 }
 
-- 
2.13.4

